### PR TITLE
IA-984 refactor LQAS-IM

### PR DESCRIPTION
**Pull request by quang-le**
## Changes

- spin all maps, charts, headers and title into their own component, moving the logic out of the `index` files
- created all new components in TS
- make these components usable by both LQAS and IM
- refactored fetching hooks
- moved all constants to IM as preparatory move to regroup LQAS and IM in one folder
- changed API so both LQAS and IM send `region_name` to front-end
- cleaned-up API a little bit
- deleted obsolete files
- deleted a lot of code
- re-organized folders

I have not moved all of LQAS yet into a single folder with IM, as some refactoring still needs to be done after the API has been updated/redesigned, and moving to the `Table`of bluesquare-components will require some tinkering and maybe its own PR as well.

Also the typing in the `utils` files is incomplete, but it can also be done in a separate issue.

### CODE REVIEW

File: plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    const [rowsPerPage, setRowsPerPage] = useState(50);
+    const [sortBy, setSortBy] = useState('asc');
+    const [sortFocus, setSortFocus] = useState<SortValues>('REGION');
+
+    const { data: lqasImData, isLoading } = useConvertedLqasImData('lqas');
+    const data = useMemo((): LqasImCampaignDataWithNameAndRegion[] => {
+        return makeDataForTable(lqasImData, campaign, round);
+    }, [lqasImData, campaign, round]);
+
+    // TODO modify sort function to sort translated messages
```
Comment (olethanh) [inline plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx:67]: it seem sortSourceKeys check the translation now?

File: plugins/polio/js/src/components/LQAS-IM/LqasImMap.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+export const LqasImMap = ({
+    type,
+    round,
+    selectedCampaign,
+    countryId,
+    campaigns,
+}) => {
+    const { formatMessage } = useSafeIntl();
+    const [renderCount, setRenderCount] = useState(0);
+    // HERE
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/LqasImMap.js:36]: ah, there was one these left 🙃

File: plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    tableKey,
+    campaign = '',
+    round,
+}) => {
+    const { formatMessage } = useSafeIntl();
+    const classes: ClassNameMap<any> = useStyles();
+    const [page, setPage] = useState(0);
+    const [rowsPerPage, setRowsPerPage] = useState(50);
+    const [sortBy, setSortBy] = useState('asc');
+    const [sortFocus, setSortFocus] = useState<SortValues>('REGION');
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx:60]: The plan is to move it to our own Table in the medium term

File: plugins/polio/js/src/components/LQAS-IM/LqasImMap.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    const { data: imData, isLoading } = useConvertedLqasImData(type);
+    const { data: shapes = [] } = useGetGeoJson(countryId, 'DISTRICT');
+
+    const scope = findScope(selectedCampaign, campaigns, shapes);
+
+    const legendItems = useMemo(() => {
+        // HERE
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/LqasImMap.js:43]: oh, 2 then...

File: plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx (hunk 1 above)
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx:67]: Yeah, maybe I should rename it

File: plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+                                                page * rowsPerPage +
+                                                    rowsPerPage,
+                                            )
+                                            .map((district, i) => {
+                                                if (district) {
+                                                    return (
+                                                        <TableRow
+                                                            key={`${tableKey}${district.name}${i}`}
+                                                            className={
+                                                                i % 2 > 0
```
Comment (beygorghor) [inline plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx:182]: FYI we have classnames module for this, but result is the same

File: plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+                                            )}
+                                        </TableRow>
+                                    </TableHead>
+                                    <TableBody>
+                                        {dataForTable
+                                            ?.slice(
+                                                page * rowsPerPage,
+                                                page * rowsPerPage +
+                                                    rowsPerPage,
+                                            )
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx:175]: good idea

File: plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+                                            .map((district, i) => {
+                                                if (district) {
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx:177]: I think I can actually delete it

File: plugins/polio/js/src/pages/LQAS/index.js [hunk 8]
```diff
@@ -183,241 +84,134 @@ export const Lqas = () => {
...
                 <Grid container item spacing={2} direction="row">
                     {campaign && (
                         <Grid item xs={12}>
                             <Box ml={2} mt={2}>
-                                <Typography variant="h5">
-                                    {formatMessage(MESSAGES.lqasPerRegion)}
-                                </Typography>
+                                <GraphTitle
+                                    text={formatMessage(MESSAGES.lqasPerRegion)}
+                                    displayTrigger={campaign}
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/pages/LQAS/index.js:115]: That's sounds about right. I will leave it as is for now, but it's probably worth changing next time we touch these components

File: plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/CaregiversTable.tsx:60]: The plan is to change it to our own `Table` in the short-medium term, but you're right

File: plugins/polio/js/src/components/LQAS-IM/DatesIgnored.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+    const datesIgnored = currentCountryName
+        ? data?.day_country_not_found[currentCountryName] ?? {}
+        : {};
+
+    return (
+        <>
+            {campaign && (
+                <>
+                    {' '}
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/DatesIgnored.tsx:27]: Nope, probably from prettier/es-lint

File: plugins/polio/js/src/components/LQAS-IM/LqasImMap.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/LqasImMap.js:43]: Actually, this whole component can be deleted 😇

**Pull request merged**